### PR TITLE
Add different window sizes in benchmarks

### PR DESCRIPTION
**Pull request by specdrake**

### CODE REVIEW

File: benchmark/Main.hs [hunk 1]
```diff
@@ -40,13 +40,11 @@ benchWithPostscanInf len name f =
 
 {-# INLINE numElements #-}
 numElements :: Int
-numElements = 100000
-
-windowSizeVal = 100
+numElements = 10000
```
Comment (harendra-kumar) [inline benchmark/Main.hs:43]: numElements can be higher than the window size, why should it be capped to 10000?

File: benchmark/Main.hs [hunk 2]
```diff
@@ -55,9 +53,15 @@ main =
...
-            , benchWithFold numElements "range" (Ring.slidingWindow windowSizeVal Statistics.range)
+            [ benchWithFold numElements "min (window size 100)" (Ring.slidingWindow 100 Statistics.min)
+            , benchWithFold numElements "min (window size 1000)" (Ring.slidingWindow 100 Statistics.min)
+            , benchWithFold numElements "min (window size 10000)" (Ring.slidingWindow 100 Statistics.min)
+            , benchWithFold numElements "max (window size 100)" (Ring.slidingWindow 100 Statistics.max)
+            , benchWithFold numElements "max (window size 1000)" (Ring.slidingWindow 100 Statistics.max)
+            , benchWithFold numElements "max (window size 10000)" (Ring.slidingWindow 100 Statistics.max)
+            , benchWithFold numElements "range (window size 100)" (Ring.slidingWindow 100 Statistics.range)
+            , benchWithFold numElements "range (window size 1000)" (Ring.slidingWindow 100 Statistics.range)
+            , benchWithFold numElements "range (window size 10000)" (Ring.slidingWindow 100 Statistics.range)
```
Comment (harendra-kumar) [inline benchmark/Main.hs:64]: Benchmarks do not match the descriptions. Same number of elements is being used in different benchmarks.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
* Lines are exceeding 80 columns.
* The benchmarks in infinite group using benchWithFoldInf and benchWithPostscanInf do not make sense anymore, they are duplicates of the large window size benchmarks and should be removed.

File: benchmark/Main.hs (hunk 2 above)
Comment (harendra-kumar) [reply benchmark/Main.hs:64]: Window size of `100` and `1000` should be good enough. We can drop the `10000` case as it takes longer and there are too many benchmarks.

File: benchmark/Main.hs (hunk 1 above)
Comment (specdrake) [reply benchmark/Main.hs:43]: ``Infinite`` benchmarks take more than 100 seconds to run with ``numElements = 100000``

File: benchmark/Main.hs [hunk 3]
```diff
@@ -67,9 +64,23 @@ main =
...
+            , benchWithPostscan numElements "max (window size 1000)"
+                (Ring.slidingWindow 1000 Statistics.max)
+            , benchWithPostscan numElements "max (window size 10000)"
+                (Ring.slidingWindow 1000 Statistics.max)
+            , benchWithPostscan numElements "range (window size 100)"
+                (Ring.slidingWindow 100 Statistics.range)
+            , benchWithPostscan numElements "range (window size 1000)"
+                (Ring.slidingWindow 1000 Statistics.range)
+            , benchWithPostscan numElements "range (window size 10000)"
+                (Ring.slidingWindow 1000 Statistics.range)
```
Comment (harendra-kumar) [inline benchmark/Main.hs:82]: Description says 10000, actual is 1000.

File: benchmark/Main.hs [hunk 4]
```diff
@@ -67,9 +64,23 @@ main =
...
+            [ benchWithPostscan numElements "min (window size 100)"
+                (Ring.slidingWindow 100 Statistics.min)
+            , benchWithPostscan numElements "min (window size 1000)"
+                (Ring.slidingWindow 1000 Statistics.min)
+            , benchWithPostscan numElements "max (window size 100)"
+                (Ring.slidingWindow 100 Statistics.max)
+            , benchWithPostscan numElements "max (window size 1000)"
+                (Ring.slidingWindow 1000 Statistics.max)
+            , benchWithPostscan numElements "max (window size 10000)"
+                (Ring.slidingWindow 1000 Statistics.max)
```
Comment (harendra-kumar) [inline benchmark/Main.hs]: Description says 10000, actual is 1000.

Comment (harendra-kumar) [reply benchmark/Main.hs]: In fact, it should be removed.

Comment (specdrake) [reply benchmark/Main.hs]: fixed

**Pull request merged**